tree/homework2.php

```diff
@@ -0,0 +1,75 @@
+<?php
+require_once ('tree_class.php');
+
+class Queue
+{
+    private $elements;
+
+    public function __construct()
+    {
+        $this->elements = array();
+    }
+
+    public function enqueue($num) {
+        array_unshift($this->elements, $num);
+    }
+
+    public function dequeue() {
+        if (!$this->isEmpty()) {
+            array_pop($this->elements);
+        }
+    }
+
+    public function front() {
+        if (!$this->isEmpty()) {
+            return $this->elements[sizeof($this->elements) - 1];
+        }
+        return null;
+    }
+
+    public function isEmpty() {
+        return empty($this->elements);
+    }
+}
+
+function inserting (object $bt, object $insert) {
+    $queue1 = new Queue;
+    if ($bt->getRoot() != null) {
+        $queue1->enqueue($bt->getRoot());
+        while (!$queue1->isEmpty()) {
+            $currNode = $queue1->front();
```

Comment:
You can extract element from Queue via method `dequeue()`. it will extract the element and remove it from the queue also.

_Notice_: you should be check if `$curNode` is `null` before doing below